### DISCUSSION TITLE
9bit SPI for RP2040 support

### DISCUSSION THREAD

**Discussion by tomobmr** [moononournation/Arduino_GFX#531] [General]
At the moment i use the SWSPI für my ST7796 TFT.

```
Arduino_DataBus *bus = new Arduino_SWSPI(GFX_NOT_DEFINED /* DC */, 9 /* CS */, 10 /* SCK */, 11 /* MOSI */, 12 /* MISO */);
Arduino_GFX *gfx = new Arduino_ST7796(bus, 13 /* RST */, 1 /* rotation */, true /* IPS */, 272, 480, 0, 0, 24, 0);
```

Unfortunately, at 480x272 pixels I only achieve a refresh rate of ~1-2 fps.

Is it planned to implement the 9bit SPI in HW for the RP2040?

**Comment by moononournation** [comment]
RP2040 not support this

**Comment by tomobmr** [comment]
![rp2040](https://github.com/user-attachments/assets/e939bdb5-a9fd-4276-90f8-65d8d6c29122)
Are you sure?

**Reply by moononournation** [reply]
I remember ESP32 is 1-512

**Reply by tomobmr** [reply]
I can't follow your words right now

**Reply by moononournation** [reply]
the ESP32 9-bit fast because it group command and data in bulk, maximum 512 bit

**Reply by tomobmr** [reply]
hm, what's about sending 1 bit and 8 bit data in succession?

**Comment by moononournation** [comment]
Can be, welcome for your contribution.
